include/bluetooth/gatt_dm.h

```diff
@@ -302,7 +302,13 @@ int bt_gatt_dm_data_release(struct bt_gatt_dm *dm);
  *
  * @param[in] dm Discovery Manager instance
  */
+#ifdef CONFIG_BT_GATT_DM_DATA_PRINT
```

Comment:
Please remove this #ifdef from here

Reply:
... if config option is not enabled you won't have `bt_gatt_dm_data_print` and linker will fail
why would you like to remove this ifdef?